File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/javascript/reverse_string.js
function reverseString(str) {
  // type your code here
  let stringToArray = [];
  let stringReversed = [];

  stringToArray = str.split('');
  // console.log(stringToArray);
  stringToArray.forEach(char => stringReversed.unshift(char));
  // console.log(stringReversed);

  return stringReversed.join('');
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));

  console.log("");

  console.log("Expecting: 'kaetseseehc'");
  console.log("=>", reverseString('cheesesteak'));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/ruby/reverse_string.rb
def reverse_string(str)
  # type your code in here
  reversed_string = ""

  str.chars.each do |char|
    reversed_string = char + reversed_string
  end

  reversed_string
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts 

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  puts 

  puts "Expecting: 'kaetseseehc'"
  puts "=>", reverse_string('cheesesteak')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution